bokehjs/src/coffee/common/embed.coffee

```diff
@@ -117,7 +117,19 @@ fill_render_item_from_script_tag = (script, item) ->
 
   logger.info("Will inject Bokeh script tag with params #{JSON.stringify(item)}")
 
-embed_items = (docs_json, render_items, websocket_url) ->
+embed_items = (docs_json_or_id, render_items, websocket_url) ->
```

Comment:
Would be nicer perhaps if we either always do it the "embed the doc as a script tag" way or the "embed the doc as JSON" way, rather than sometimes doing one and sometimes doing the other.

It looks like you added the "embed doc as a script" way in order to scrape a document out of an HTML file?

If so I would say we should always go from `Application` to PNG/SVG, rather than scrape-json-from-html-file to PNG/SVG. The Application could be spelled as a JSON file, which gives you the JSON case (add application/spellings/json.py which does `Document.from_json_string` and you have that).

Command line implications:
 * `bokeh json myapp.py` - dumps myapp.py as JSON (creates app document and saves `document.to_json_string()`)
 * `bokeh html myapp.json` - sticks the JSON in an HTML file (needs a json spelling handler)
 * `bokeh png myapp.py`, `bokeh png myapp.json` - output the app from whatever format to PNG

I know you aren't done with the PR so maybe you've already gone in another direction.